balancer-js/src/modules/pools/model-provider.ts

```diff
@@ -0,0 +1,124 @@
+import type { BalancerDataRepositories, Pool, PoolModel } from '@/types';
+import { PoolApr } from './apr/apr';
+import { Liquidity } from '../liquidity/liquidity.module';
+
+// Can be changed outside for dependent services rate-limits
+// eslint-disable-next-line prefer-const
+export let POOLS_PER_PAGE = 10;
+
+/**
+ * Use-cases layer for generating live pools data
+ */
+export class ModelProvider {
+  constructor(private repositories: BalancerDataRepositories) {}
+
+  static async resolve(model: PoolModel): Promise<Pool> {
+    return {
+      ...model,
```

Comment:
With the way this is implemented if a user updates the pool then calls apr() again it won't use the updated data, because it's copying the properties of model into this Pool object and then the apr function is using the original data, not the changed data saved to the Pool.

The wrap function has the same issue.

Reply:
True, it's not mutable. For now let's use `this` but the problem lies in the data store not being reactive. State can only be updated upstream, onchain, or in case of pricing time based. Here you can update state directly on the model, but it's unexpected and repository will return stale data, because it's cached.

I'm thinking about a next PR for making data reactive. Something like syncing via websockets, and instantiate new model class from that.